File: sql/postgresql/Dart/PostgreSQLParserBase.dart
import 'dart:convert';
import 'package:antlr4/antlr4.dart';

abstract class PostgreSQLParserBase extends Parser {

  PostgreSQLParserBase(TokenStream input) : super(input);

  void ParseRoutineBody() {
  }

  bool OnlyAcceptableOps() {
    var token = (this.inputStream as CommonTokenStream).LT(1)!;
    var text = token.text;
    return text == '!' || text == '!!' || text == '!=-';
  }
}

File: sql/postgresql/Python3/PostgreSQLParserBase.py
from antlr4 import *

relativeImport = False
if __name__ is not None and "." in __name__:
    relativeImport = True

class PostgreSQLParserBase(Parser):
    def ParseRoutineBody(self):
        return

    def OnlyAcceptableOps(self):
        c = self._input.LT(1)
        text = c.text
        return text == "!" or text == "!!" or text == "!=-"
